### PR TITLE
rebaseline_tests.py: Add --new-branch and --clear. NFC

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: tools/maint/rebaseline_tests.py [hunk 1]
```diff
@@ -64,8 +64,13 @@ def process_changed_file(filename):
 def main(argv):
   parser = argparse.ArgumentParser()
   parser.add_argument('-s', '--skip-tests', action='store_true', help="don't actually run the tests, just analyze the existing results")
+  parser.add_argument('-b', '--new-branch', action='store_true', help='create a new branch containing the updates')
+  parser.add_argument('-c', '--clear-cache', action='store_true', help='clear the cache before rebaselining (useful when working with llvm changes)')
```
Comment (kripken) [inline tools/maint/rebaseline_tests.py:68]: Should this be the default perhaps?

File: tools/maint/rebaseline_tests.py [hunk 2]
```diff
@@ -91,14 +96,18 @@ def main(argv):
...
+```
 '''
 
   for file in filenames:
     commit_message += process_changed_file(file)
 
   commit_message += f'\nAverage change: {statistics.mean(all_deltas):+.2f}% ({min(all_deltas):+.2f}% - {max(all_deltas):+.2f}%)\n'
 
-  run(['git', 'checkout', '-b', 'rebaseline_tests'])
+  commit_message += "'''\n"
```
Comment (kripken) [inline tools/maint/rebaseline_tests.py:107]:
```suggestion
  commit_message += "```\n"
```

Comment (kripken) [reply tools/maint/rebaseline_tests.py:107]: (i think this was a typo? this change should address @aheejin 's point above)

File: tools/maint/rebaseline_tests.py (hunk 1 above)
Comment (sbc100) [reply tools/maint/rebaseline_tests.py:68]: I think most of the time this script will be used to update expectations in response for local emscripten changes (i.e. minor changes to JS libraries) not upstream llvm change.   At least for me I often run rebases without clearing the cache when I know llvm has not changed.

Comment (kripken) [reply tools/maint/rebaseline_tests.py:68]: Fair enough. lgtm either way.

File: tools/maint/rebaseline_tests.py [hunk 3]
```diff
@@ -91,14 +96,18 @@ def main(argv):
 The following ({len(filenames)}) test expectation files were updated by
 running the tests with `--rebaseline`:
 
+```
```
[the remark this answers is not in the document]
Comment (sbc100) [reply tools/maint/rebaseline_tests.py:99]: Good catch!

**Pull request merged**